turbopack/crates/turbopack-ecmascript/src/simple_tree_shake/mod.rs

```diff
@@ -0,0 +1,89 @@
+//! Intermediate tree shaking that uses global information but not good as the full tree shaking.
+
+use anyhow::{bail, Context, Result};
+use rustc_hash::{FxHashMap, FxHashSet};
+use turbo_rcstr::RcStr;
+use turbo_tasks::{ResolvedVc, TryJoinIterExt, Vc};
+use turbopack_core::{
+    module_graph::{ModuleGraph, SingleModuleGraph},
+    resolve::ExportUsage,
+};
+
+use crate::chunk::EcmascriptChunkPlaceable;
+
+pub async fn is_export_used(
+    graph: ResolvedVc<ModuleGraph>,
+    module: ResolvedVc<Box<dyn EcmascriptChunkPlaceable>>,
+    export_name: RcStr,
+) -> Result<bool> {
```

Comment:
1. Ideally this would be a turbotask, can you try changing this into a turbo task but with the signature `used_exports(graph, module) -> Vc<Set<Export>>`. That should reduce the number of tasks by an order of magnitude.

1. I would change this function into a method on `ExportUsageInfo`, that's what the API is on the other global information function as well. e.g. https://github.com/vercel/next.js/blob/68c3e13ab52e128c39ab40a11543390b42850a48/turbopack/crates/turbopack-core/src/module_graph/async_module_info.rs#L22